controllers/azurecluster_reconciler.go

```diff
@@ -159,21 +159,22 @@ func (s *azureClusterService) Delete(ctx context.Context) error {
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "controllers.azureClusterService.Delete")
 	defer done()
 
-	if !ShouldDeleteIndividualResources(ctx, s.scope) {
+	groupSvc, err := GetService(groups.ServiceName, s.services)
```

Comment:
I'm leaning toward keeping this as-is because it's not a huge gain but would require refactoring a bunch of unit tests and I already went down that road once when I was trying to have the function itself assert the type (ended up backing out) and that was not a good time. Also the function is used for vnetPeeringSvc too and I could see other usages in the future so it wouldn't make as much sense if it's more than 1-2 services being "special".